luda-adventure/src/app/game/types/game_object/types/positioner/predict_collision.rs

```diff
@@ -121,22 +121,43 @@ fn predict_intersection_on_one_axis(
     target_less_side: Tile,
     target_greater_side: Tile,
 ) -> IntersectionPrediction {
+    enum MovementDirection {
+        Positive,
+        Zero,
+        Negative,
+    }
+    let movement_direction = match (speed * 1.0.ms())
+        .partial_cmp(&0.0_f32.tile())
+        .expect("Invalid speed")
+    {
+        std::cmp::Ordering::Less => MovementDirection::Negative,
+        std::cmp::Ordering::Equal => MovementDirection::Zero,
+        std::cmp::Ordering::Greater => MovementDirection::Positive,
+    };
     let target_greater_side_minus_character_less_side = target_greater_side - character_less_side;
     let target_greater_side_minus_character_less_side_divided_by_speed =
         speed.invert() * target_greater_side_minus_character_less_side;
     let target_less_side_minus_character_greater_side = target_less_side - character_greater_side;
     let target_less_side_minus_character_greater_side_divided_by_speed =
         speed.invert() * target_less_side_minus_character_greater_side;
```

Comment:
I don't feel good with this code because these variable names doesn't have any meaning but just the right side's calculation.

Let's think about `let one_plus_two = 1 + 2;`. Why it need to make a new variable? Any good reason for it?

I think variable names are meaningful when it has different meaning against right-side calculation.

For example
```rust
let target_greater_side_minus_character_less_side = target_greater_side - character_less_side;
// would be better with
let distance_between_character_less_to_target_greater = target_greater_side - character_less_side;
```

Reply:
Ahh, I agree

Reply:
[80befd7](https://github.com/NamseEnt/namseent/pull/359/commits/80befd7b6f487deec5db35fec423c75dbc814ec4)